lib/hotel_dispatcher.rb

```diff
@@ -0,0 +1,95 @@
+require_relative 'room'
+
+module Hotel
+  class HotelDispatcher
+    attr_accessor :reservations, :rooms
+
+    def initialize
+
+      @reservations = []
+      create_rooms
+    end 
+
+    # Wave 1 - Create 20 rooms. 
+    def create_rooms
+      @rooms = []
+        (1..20).each do |room_num|
+          room = Hotel::Room.new(room_num: 1)
+          room.room_num = room_num
+          @rooms << room
+        end
+        return @rooms
+    end
+
+    # Wave 1 - Make a new reservation.
+    def make_reservation(start_date, end_date)
+      if end_date < start_date
+        raise ArgumentError
+      end
+      room = create_rooms[0]
+      new_reservation = Hotel::Reservation.new(start_date: start_date, end_date: end_date, room: room)
+      @reservations << new_reservation
+      return @reservations
+    end 
+
+    # Wave 1 - Find all reservations for a given room and date range.
+    def find_all_res_for_room(room_num, start_date, end_date)
+      found_reservations = []
+        @reservations.each  do |reservation| 
+          next if reservation.room.room_num != room_num
+            if ((start_date > reservation.start_date && start_date < reservation.end_date) ||
+            (end_date > reservation.start_date && start_date < reservation.end_date)) || 
+            (reservation.start_date < start_date && reservation.end_date > end_date)
```

Comment:
This looks slightly different from what I see on lines 64-67 and I'm unsure why. This is a good candidate for code that would be appropriate to pull out into it's own method (probably most appropriately a method in the Reservation class).